src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -173,6 +177,111 @@ Task<TResult> IInterleavingContext.CallSubOrchestratorWithRetryAsync<TResult>(st
+
+                DurableHttpRequest durableAsyncHttpRequest = this.CreateNewHttpRequestMessageForAsyncResponse(req);
+                durableAsyncHttpRequest = AddLocationHeaderToAsyncRequest(durableAsyncHttpRequest, durableHttpResponse);
+
+                durableHttpResponse = await this.ScheduleDurableHttpActivityFunctionAsync(durableAsyncHttpRequest);
+
+                currStatusCode = durableHttpResponse.StatusCode;
+            }
+
+            return durableHttpResponse;
+        }
+
+        private async Task<DurableHttpResponse> ScheduleDurableHttpActivityFunctionAsync(DurableHttpRequest req)
+        {
+            string serializedRequest = MessagePayloadDataConverter.HttpConverter.Serialize(req);
+            JToken durableHttpResponseJson = await this.CallDurableTaskFunctionAsync<JToken>(
+                functionName: HttpOptions.HttpTaskActivityReservedName,
+                functionType: FunctionType.Activity,
+                oneWay: false,
+                instanceId: null,
+                operation: null,
+                retryOptions: null,
+                input: serializedRequest);
+
+            JsonSerializer serializer = new JsonSerializer();
+            serializer.Converters.Add(new DurableHttpResponseJsonConverter(typeof(DurableHttpResponse)));
+            DurableHttpResponse durableHttpResponse = durableHttpResponseJson.ToObject<DurableHttpResponse>(serializer);
+            return durableHttpResponse;
+        }
+
+        private static IDictionary<string, StringValues> CreateStringValuesHeaderDictionary(IList<KeyValuePair<string, string>> headers)
+        {
+            IDictionary<string, StringValues> newHeaders = new Dictionary<string, StringValues>();
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    if (newHeaders.ContainsKey(header.Key))
+                    {
+                        StringValues values = StringValues.Concat(header.Value, newHeaders[header.Key]);
+                        newHeaders[header.Key] = values;
+                    }
+                    else
+                    {
+                        StringValues stringValues = new StringValues(header.Value);
+                        newHeaders.Add(header.Key, stringValues);
+                    }
+                }
+            }
+
+            return newHeaders;
+        }
+
+        private static DurableHttpRequest AddLocationHeaderToAsyncRequest(DurableHttpRequest durableHttpRequest, DurableHttpResponse durableHttpResponse)
+        {
+            IDictionary<string, StringValues> durableHttpRequestHeaders = durableHttpResponse.Headers;
+            string locationValue = durableHttpRequestHeaders["Location"];
+            durableHttpRequest.Uri = new Uri(locationValue);
+            return durableHttpRequest;
+        }
+
+        private DurableHttpRequest CreateNewHttpRequestMessageForAsyncResponse(DurableHttpRequest durableHttpRequest)
```

Comment:
When I read this method name, I think that it will take in a DurableHttpResponse object and return a `DurableHttpRequest` object. When I read the body of the method, it looks like it is just cloning the original request. I would change the name to reflect the body in this scenario.